File: mobile-client/android/UseCase/Shipment/Tracking/ListShipmentsUseCase.kt
package com.smartresponsor.mobile.client.usecase.shipment.tracking

import com.smartresponsor.mobile.client.contract.shipment.tracking.ListShipmentsQuery
import com.smartresponsor.mobile.client.contract.shipment.tracking.ShipmentTrackingSummary
import com.smartresponsor.mobile.client.data.shipment.tracking.ShipmentTrackingGateway

/**
 * Marketing America Corp. Oleksandr Tishchenko
 */
class ListShipmentsUseCase(
    private val shipmentTrackingGateway: ShipmentTrackingGateway,
) {
    suspend operator fun invoke(query: ListShipmentsQuery): List<ShipmentTrackingSummary> =
        shipmentTrackingGateway.listShipments(query)
}

File: client/ios/UseCase/Shipment/Tracking/ListShipmentsUseCase.swift
import Foundation

// Marketing America Corp. Oleksandr Tishchenko
struct ListShipmentsUseCase {
    let shipmentTrackingGateway: ShipmentTrackingGateway

    func callAsFunction(query: ListShipmentsQuery) async throws -> [ShipmentTrackingSummary] {
        try await shipmentTrackingGateway.listShipments(query: query)
    }
}
